src/handshake/server.rs

```diff
@@ -92,14 +94,23 @@ impl<'a, T: AsyncRead + AsyncWrite + Unpin> Server<'a, T> {
         loop {
             crate::read(&mut self.socket, &mut self.buffer, BLOCK_SIZE).await?;
 
+            let limit = std::cmp::min(self.buffer.len(), MAX_HEADERS_SIZE);
+
             // We don't expect body, so can search for the CRLF headers tail from
             // the end of the buffer.
-            if self.buffer[skip..].windows(4).rev().find(|w| w == b"\r\n\r\n").is_some() {
+            if self.buffer[skip..limit].windows(4).rev().find(|w| w == b"\r\n\r\n").is_some() {
```

Comment:
Clippy suggests `any` instead of `find` and a few other things.

Reply:
I used `any` in the CORS branch, not sure why this didn't occur to me here.